### PR TITLE
[various] allow to define a default mail template

### PR DESCRIPTION
**Pull request by dka-odoo**
Task - https://www.odoo.com/web?#id=32882&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720

pad - https://pad.odoo.com/p/r.517c3cc7b8dfaa42b570b773c33c808b

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/wizard/mail_compose_message.py [hunk 1]
```diff
@@ -65,6 +65,7 @@ def default_get(self, fields):
         result['model'] = result.get('model', self._context.get('active_model'))
         result['res_id'] = result.get('res_id', self._context.get('active_id'))
         result['parent_id'] = result.get('parent_id', self._context.get('message_id'))
+        result['template_id'] = self.env['ir.values'].sudo().get_default('mail.compose.message', 'template_id', condition='mail_compose_act_btn=' + (self.env.context.get('mail_compose_act_btn') or '') + '&' + 'model=' + (result['model'] or '')) or result.get('template_id')
```
Comment (rim-odoo) [inline addons/mail/wizard/mail_compose_message.py]: `ir.values` are now called `ir.default`

File: addons/mail/wizard/mail_compose_message.py [hunk 2]
```diff
@@ -125,6 +126,16 @@ def _get_composition_mode_selection(self):
     message_type = fields.Selection(default="comment")
     subtype_id = fields.Many2one(default=lambda self: self.sudo().env.ref('mail.mt_comment', raise_if_not_found=False).id)
 
+    is_default_template = fields.Boolean(compute='_compute_is_default_template')
+
+    @api.depends('template_id')
+    def _compute_is_default_template(self):
+        IrValues = self.env['ir.values'].sudo()
+        for compose in self:
+            template_id = IrValues.get_default('mail.compose.message', 'template_id', condition='mail_compose_act_btn=' + (self.env.context.get('mail_compose_act_btn') or '') + '&' + 'model=' + (compose.model or ''))
```
Comment (rim-odoo) [inline addons/mail/wizard/mail_compose_message.py]: for code readability, maybe extract the computing of `condition=` to an helper function

File: addons/mail/wizard/mail_compose_message.py (hunk 1 above)
Comment (rim-odoo) [reply addons/mail/wizard/mail_compose_message.py]: for code readability, maybe extract the computing of `condition=` to an helper function

File: addons/mail/wizard/mail_compose_message.py [hunk 3]
```diff
@@ -332,6 +343,12 @@ def get_mail_values(self, res_ids):
             results[res_id] = mail_values
         return results
 
+    @api.multi
+    def mark_template_as_default(self):
+        self.ensure_one()
+        if self.template_id and self.env.context.get('mail_compose_act_btn'):
+            self.env['ir.values'].sudo().set_default('mail.compose.message', "template_id", self.template_id.id, condition="mail_compose_act_btn=" + (self.env.context.get('mail_compose_act_btn')) + '&' + 'model=' + (self.model or ''))
```
Comment (rim-odoo) [inline addons/mail/wizard/mail_compose_message.py]: same issue with `ir.values` here

File: addons/mail/wizard/mail_compose_message.py (hunk 2 above)
Comment (rim-odoo) [reply addons/mail/wizard/mail_compose_message.py]: same issue with `ir.values` here

**Comment by tde-banana-odoo** [comment]
Task canceled.

**Pull request closed without merging**